test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/https/WorkbenchHttpsIntegrationTest.java

```diff
@@ -102,51 +100,62 @@ public void testLoginScreen() throws InterruptedException {
-                                if (organizationalUnit.get("name").getAsString().equals(ORGANIZATION_UNIT_NAME)) {
-                                    found = true;
-                                }
-                            }
-                        }
-
-                        Assertions.assertThat(found).isTrue();
-                    }
-                }
+                Assertions.assertThat(organizationalUnitExists(ORGANIZATION_UNIT_NAME, httpClient)).isTrue();
             }
         } catch (Exception e) {
             Assertions.fail("Unable to connect to workbench REST API", e);
         }
     }
 
+    private void createOrganizationalUnit(String name, CloseableHttpClient httpClient) {
+        final HttpPost createOrgUnitRequest;
+        try {
+            final URL url = new URL(workbenchWithKieServerScenario.getWorkbenchDeployment().getSecureUrl(), organizationalUnitRestRequest);
+            createOrgUnitRequest = new HttpPost(url.toString());
+            createOrgUnitRequest.setHeader("Content-Type", "application/json");
+            createOrgUnitRequest.setEntity(new StringEntity(createOrganizationalUnitJson(name)));
+        } catch (Exception e) {
+            throw new RuntimeException("Error in creating rest request for new organizational unit", e);
+        }
+
+        try (CloseableHttpResponse response = httpClient.execute(createOrgUnitRequest)) {
+            Assertions.assertThat(response.getStatusLine().getStatusCode()).isEqualTo(HttpsURLConnection.HTTP_ACCEPTED);
+        } catch (Exception e) {
+            Assertions.fail("Error creating new organizational unit", e);
+        }
+    }
+
+    private boolean organizationalUnitExists(String name, CloseableHttpClient httpClient) {
+        final HttpGet getOrgUnitRequest;
+        try {
+            final URL url = new URL(workbenchWithKieServerScenario.getWorkbenchDeployment().getSecureUrl(), organizationalUnitRestRequest);
+            getOrgUnitRequest = new HttpGet(url.toString());
+            getOrgUnitRequest.setHeader("Content-Type", "application/json");
+        } catch (Exception e) {
+            throw new RuntimeException("Error in creating request for list of organizational units", e);
+        }
+        try (CloseableHttpResponse response = httpClient.execute(getOrgUnitRequest)) {
+            Assertions.assertThat(response.getStatusLine().getStatusCode()).isEqualTo(HttpsURLConnection.HTTP_OK);
+            try (InputStream inputStream = response.getEntity().getContent()) {
+                String responseContent = IOUtils.toString(inputStream, "UTF-8");
+                Assertions.assertThat(responseContent).isNotEmpty();
+
+                Gson gson = new Gson();
+                return StreamSupport.stream(gson.fromJson(responseContent, JsonArray.class).spliterator(), false)
+                        .anyMatch(x -> x.getAsJsonObject().get("name").getAsString().equals(name));
```

Comment:
This could be made slightly more readable if it were split into independent steps. Such as this:
`.map(x -> x.getAsJsonObject()).map(o -> o.get("name").getAsString()).anyMatch(String::equals);`

Even better if each call is on its own line.